core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala

```diff
@@ -1566,6 +1566,56 @@ class WorkspaceApiServiceSpec extends ApiServiceSpec {
       }
   }
 
+  it should "return 204 for a user with read access on an unlocked workspace" in withTestDataApiServicesAndUser("reader-access") { services =>
+    Get(s"${testData.workspace.path}/checkIamActionWithLock/read") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.NoContent) {
+          status
+        }
+      }
+  }
+
+  it should "return 403 for a reader testing the write action on an unlocked workspace" in withTestDataApiServicesAndUser("reader-access") { services =>
+    Get(s"${testData.workspace.path}/checkIamActionWithLock/write") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.Forbidden) {
+          status
+        }
+      }
+  }
+
+  it should "return 204 for a reader testing the read action on a locked workspace" in withTestDataApiServicesAndUser("reader-access") { services =>
+    Get(s"${testData.workspaceLocked.path}/checkIamActionWithLock/read") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.NoContent) {
+          status
+        }
+      }
+  }
+
+  it should "return 204 for a writer testing the write action on an unlocked workspace" in withTestDataApiServicesAndUser("writer-access") { services =>
+    Get(s"${testData.workspace.path}/checkIamActionWithLock/write") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.NoContent) {
+          status
+        }
+      }
+  }
+
+  it should "return 403 for a writer testing the write action on a locked workspace" in withTestDataApiServicesAndUser("writer-access") { services =>
+    Get(s"${testData.workspaceLocked.path}/checkIamActionWithLock/write") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.Forbidden) {
+          status
+        }
+      }
+  }
+
```

Comment:
I'm curious - are you just adding coverage here? Was there a specific issue that needed testing?

Reply:
This is restoring some coverage that was lost as a result of the changes in this PR. In the `ServiceSpec` we were testing response codes and response data. Now that the service layer isn't returning response codes anymore (except in the case of some exceptions), I needed to add that coverage in the `ApiSpec`